local/ntap/workflows-560-rnaseq-human.py

```diff
@@ -0,0 +1,213 @@
+    def synindex_run_name(self) -> str:
+        """The name of the synindex run."""
+        return f"synindex_{self.id}_{self.run_number}"
+
+
+def generate_datasets() -> list[Dataset]:
+    """Generate list of datasets.
+
+    Source: https://sagebionetworks.jira.com/browse/WORKFLOWS-538
+
+    This is expecting that the SampleSheet to be run exists in the `prefix` location.
+    """
+    return [
+        Dataset(
+            id="syn53972620",
+            samplesheet="syn51198895_RNA_Seq_SampleSheet.csv",
+            staging_key="samplesheets/RNASeq_Process/WORKFLOWS-560/",
+            bucket_name="ntap-add5-project-tower-bucket",
+            synapse_id_for_output="syn55228511",
+            run_number=1,
+            output_number=1,
+        )
+    ]
+
+def stage_samplesheet(syn: Synapse, dataset: Dataset):
+    """Download the samplesheet from synapse and upload it to S3 in the location where synstage
+    is going to grab the file.
+
+    :param syn: The logged in synapse instance.
+    :param dataset: The dataset to stage the samplesheet for.
+    """
+    samplesheet_file = syn.get(dataset.id)
+    samplesheet_file_path = samplesheet_file.path
+
+    s3.upload_file(
+        samplesheet_file_path, dataset.bucket_name, dataset.samplesheet_to_stage_key
+    )
+
+
+def prepare_synstage_info(dataset: Dataset) -> LaunchInfo:
+    """Generate LaunchInfo for nf-synstage."""
+    return LaunchInfo(
+            run_name=dataset.synstage_run_name,
+            pipeline="Sage-Bionetworks-Workflows/nf-synapse",
+            revision="main",
+            profiles=["sage"],
+            entry_name="NF_SYNSTAGE",
+            params={
+                "input": dataset.samplesheet_location,
+                "outdir": dataset.staging_location,
+            },
+            workspace_secrets=["SYNAPSE_AUTH_TOKEN"]
+        )
+    
+def prepare_rnaseq_launch_info(dataset: Dataset) -> LaunchInfo:
+    """Generate LaunchInfo for nf-core/rnaseq workflow run."""
+    return LaunchInfo(
+        run_name=dataset.rnaseq_run_name,
+        pipeline="nf-core/rnaseq",
+        revision="3.11.2",
+        profiles=["sage"],
+        params={
+            "input": dataset.staged_samplesheet_location,
+            "outdir": dataset.output_directory,
+            "gencode": True,
+            "fasta": "s3://sage-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Sequence/WholeGenomeFasta/genome.fa",
+            "gtf": "s3://sage-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Annotation/Genes/genes.gtf",
+            "gene_bed": "s3://sage-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Annotation/Genes/genes.bed",
+            "star_index": "s3://sage-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Sequence/STARIndex/",
+            # TODO: These could not be found in `s3://sage-igenomes/igenomes/Homo_sapiens/`
+            # "transcript_fasta": "",
+            # "salmon_index": "",
+            # "rsem_index": "",
+        },
+        pre_run_script="NXF_VER=23.10.0",
```

Comment:
@BWMac If I recall what you had said, we shouldn't we specifying this manually?

Reply:
We talked about the nextflow version for `synstage` last meeting. I confirmed with Brad that we can keep using the specified version for other steps for this round. But we can open a new ticket to update this part later.